components/consent-mgt/org.wso2.carbon.identity.consent.mgt/src/main/java/org/wso2/carbon/identity/consent/mgt/listener/TenantConsentMgtListener.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.consent.mgt.listener;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.consent.mgt.core.exception.ConsentManagementException;
+import org.wso2.carbon.consent.mgt.core.model.PurposeCategory;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.consent.mgt.internal.IdentityConsentDataHolder;
+import org.wso2.carbon.identity.core.AbstractIdentityTenantMgtListener;
+import org.wso2.carbon.stratos.common.beans.TenantInfoBean;
+import org.wso2.carbon.stratos.common.exception.StratosException;
+
+/**
+ * Tenant listner which adds default purpose category for a tenant
+ */
+public class TenantConsentMgtListener extends AbstractIdentityTenantMgtListener {
+
+    private static final Log log = LogFactory.getLog(TenantConsentMgtListener.class);
+    private static final String DEFAULT_PURPOSE_CATEGORY = "DEFAULT";
+
+    @Override
+    public void onTenantCreate(TenantInfoBean tenantInfoBean) throws StratosException {
+
+        if (isSsoConsentManagementEnabled()) {
```

Comment:
why do we need to check this? What happens if this config enables after creating the tenant. Also, how do we support this for existing tenants?

Reply:
Then they have to add this through API. Basically when this config is disabled nothing related to consent functionalities will work. Hence no additional DB queries.

Reply:
This is fine if we have a UI. Since we don't have a UI to add purpose-category, it is wrong to ask to do an api call to add default purpose/purpose category.

It is fine.. if this create on the fly when DEFAULT category is required in SSO flow. @omindu can u confirm this?

Reply:
Yes.. Default category is created on the fly if it's not there.

Reply:
ok. can we do this on tenant loading? Then this will work for existing tenants as well.

Reply:
This is even costlier. What we planned was no CM db queries if CM is disabled system wide

Reply:
yes. you can do within if (isSsoConsentManagementEnabled()) {

Reply:
Do you mean on tenantActivation ? There is no explicit listener for tenant loading

Reply:
ok